portfolio/src/main/webapp/script.js

```diff
@@ -35,9 +35,57 @@ async function helloName() {
   });
 }
 
+/** Fetches comments from the server and adds them to the DOM*/
+async function loadComments() {
+  let maxComments = parseMaxComments();
+  let url = '/data?maxComments=' + maxComments;
+  fetch(url).then(response => response.json()).then((comments) => {
+    const commentsListElement = document.getElementById('comments-list');
+    comments.forEach((comment) => {
+      commentsListElement.appendChild(createCommentElement(comment));
+    })
+  });
+}
+
+function parseMaxComments() {
+  let currentUrl = getURL();
+  let parser = new URL(currentUrl); 
+  let queryString = parser.search;
+  let pos = queryString.search("maxComments=");
+  let maxComments
+  if(pos != -1) {
```

Comment:
Could you comment what this is doing? It's a bit unclear why 12 would be the number here. Setting default max comments as a constant value would also help clarify this.

Reply:
I add 12 b/c the length of "maxComments=" is 12, and pos returns the beginning index of "maxComments=". I'll add a clarifying comment

Reply:
Yeah, please add some clarification.